modules/firebase/firebase_test.go

```diff
@@ -0,0 +1,74 @@
+package firebase_test
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/firestore"
+	"github.com/stretchr/testify/require"
+
+	"github.com/testcontainers/testcontainers-go"
+
+	"github.com/testcontainers/testcontainers-go/modules/firebase"
+)
+
+func TestFirebase(t *testing.T) {
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(3*time.Minute))
+	defer cancel()
+
+	ctr, err := firebase.Run(ctx, "ghcr.io/u-health/docker-firebase-emulator:13.29.2",
+		firebase.WithRoot(filepath.Join("testdata", "firebase")),
+	)
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+
+	// Ports are linked to the example config in firebase/firebase.json
+
+	firestoreURL, err := ctr.ConnectionString(ctx, "8080/tcp")
+	require.NoError(t, err)
+	require.NotEmpty(t, firestoreURL)
+
+	t.Setenv("FIRESTORE_EMULATOR_HOST", firestoreURL)
+	c, err := firestore.NewClient(ctx, firestore.DetectProjectID)
+	require.NoError(t, err)
+	defer c.Close()
+
+	w, err := c.Collection("example").Doc("one").Set(ctx, map[string]any{
+		"foo": "bar",
+	})
+	require.NoError(t, err)
+	require.NotNil(t, w)
+
+	snap, err := c.Collection("example").Doc("one").Get(ctx)
+	require.NoError(t, err)
+	var out map[string]string
+	err = snap.DataTo(&out)
+	require.NoError(t, err)
+	require.Equal(t, "bar", out["foo"])
+}
+
+func TestFirebaseBadDirectory(t *testing.T) {
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(3*time.Minute))
+	defer cancel()
+
+	ctr, err := firebase.Run(ctx, "ghcr.io/u-health/docker-firebase-emulator:13.29.2",
+		firebase.WithRoot(filepath.Join("testdata", "failure")),
+	)
+	// In this case, the file gets copied over at /srv/failure (instead of /srv/firebase)
+	// and this stops working.
+	// What would be a solution here? Previously I just added a check that the root must
+	// end in "/firebase"... I could do the same.
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
```

Comment:
bug: this should fail, as I understand it this is down to how our copy works or doesn't. I have an unfinished PR in this [area here](https://github.com/testcontainers/testcontainers-go/pull/2782).

I would change this to check for err as expected and mark the test as skip for now.